tests/auth_tests/test_tokens.py

```diff
@@ -39,23 +39,24 @@ def test_timeout(self):
         # Uses a mocked version of PasswordResetTokenGenerator so we can change
         # the value of 'now'.
         user = User.objects.create_user('tokentestuser', 'test2@example.com', 'testpw')
+        now = datetime.now()
         p0 = PasswordResetTokenGenerator()
```

Comment:
Should the initial token generator not also use the mocked now to guarantee exact calculations? Above, any time that passes between the `now =` and the actual generation of the token will be *extra* lifetime of the token (which can still make the tests fail by keeping the token valid in the timeout + 1 test if this time is >= 1 second).

Reply:
Good point :+1: